src/hotspot/share/runtime/os.cpp

```diff
@@ -205,8 +205,8 @@ char* os::iso8601_time(jlong milliseconds_since_19700101, char* buffer, size_t b
   // Print an ISO 8601 date and time stamp into the buffer
   const int year = 1900 + time_struct.tm_year;
   const int month = 1 + time_struct.tm_mon;
-  const int printed = jio_snprintf(buffer, buffer_length,
-                                   "%04d-%02d-%02dT%02d:%02d:%02d.%03d%c%02d%02d",
+  const int printed = os::snprintf(buffer, buffer_length,
+                                   "%04d-%02d-%02dT%02d:%02d:%02d.%03d%c%02ld%02ld",
```

Comment:
This is making the unwarranted assumption that `time_t` == `long`.  Also, the "l" length modifier should
probably never be used in shared code, just as the `long` type shouldn't be used, because of the LP vs LLP
issue.  Better would be to declare `zone_hours` and `zone_min` to be `int`, and explicitly convert the initial
value expressions to `int`.

Reply:
Fixed.

Reply:
Both are wrong. You can use `%02d` but **must** ***explicitly*** cast to `int` (or to `long` if you use `%02ld` but that makes no sense for a two-digit number) because `time_t` is often wider than `long`… which `-Wformat` would have shown.

Reply:
Ah, nevermind, I see now that the PR was changed to change the variable type instead… which is fair in this situation.